src/Compilers/CSharp/Test/Semantic/Semantics/NullableContextTests.cs

```diff
@@ -413,8 +429,894 @@ void verify(CSharpParseOptions parseOptions, bool expectedFlowState, params stri
             }
         }
 
-        private static string[] GetNullableDataKeysAsStrings(ConcurrentDictionary<object, int> nullableData) =>
-            nullableData.Keys.Select(key => GetNullableDataKeyAsString(key)).OrderBy(key => key).ToArray();
+        private struct NullableDirectives
+        {
+            internal readonly string[] Directives;
+            internal readonly NullableContextState.State ExpectedWarningsState;
+            internal readonly NullableContextState.State ExpectedAnnotationsState;
+
+            internal NullableDirectives(string[] directives, NullableContextState.State expectedWarningsState, NullableContextState.State expectedAnnotationsState)
+            {
+                Directives = directives;
+                ExpectedWarningsState = expectedWarningsState;
+                ExpectedAnnotationsState = expectedAnnotationsState;
+            }
+        }
+
+        [Fact]
+        [WorkItem(49746, "https://github.com/dotnet/roslyn/issues/49746")]
+        public void AnalyzeMethodsInEnabledContextOnly_01()
```

Comment:
Nit: Did you consider using a `[Theory]` instead of a `[Fact]` here? That would make the tests a bit more debuggable when they fail. Not a big issue though.

Reply:
> Did you consider using a `[Theory]` instead of a `[Fact]` here?

I did consider it, and I was on the fence. In the end, it seemed simpler to express the combinations inline in the test method.

---
In reply to: [545954318](https://github.com/dotnet/roslyn/pull/49953#discussion_r545954318) [](ancestors = 545954318)

Reply:
Please reconsider this. Failures are going to be very hard to understand in the current form.

---
In reply to: [545974860](https://github.com/dotnet/roslyn/pull/49953#discussion_r545974860) [](ancestors = 545974860,545954318)

Reply:
Either that, or modify the assertions to include the current options under test.

---
In reply to: [546883437](https://github.com/dotnet/roslyn/pull/49953#discussion_r546883437) [](ancestors = 546883437,545974860,545954318)

Reply:
Using a `[Theory]` attribute or modifying the assertions feels unnecessary here. I'd prefer to keep this simpler.

---
In reply to: [546884705](https://github.com/dotnet/roslyn/pull/49953#discussion_r546884705) [](ancestors = 546884705,546883437,545974860,545954318)

Reply:
Then please modify the asserts to assert what is currently being tested. Tests like these, where the assert tells you nothing about with of the 16 cases is being tested, are a serious frustration of mine with our codebase, and I'd prefer if we didn't add to the list.

---
In reply to: [548388574](https://github.com/dotnet/roslyn/pull/49953#discussion_r548388574) [](ancestors = 548388574,546884705,546883437,545974860,545954318)